### PR TITLE
Suheyb - Week3

### PR DESCRIPTION
**Pull request by suh3yb**

### CODE REVIEW

File: Week3/app.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
+'use strict';
+
+const express = require('express');
+const app = express();
+
+const { execOneArgQuery, execMultiArgQuery, queries, renderReadme } = require('./src');
+
+const PORT = 3000;
+const USERS_SLUG = 'users';
```
Comment (gajduk) [inline Week3/app.js]: smart that you defined these on top 👍

File: Week3/src/isInputValid.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const { ErrorWithStatus } = require('./error');
+
+const USERS_REG = /^\/users$/i;
```
Comment (gajduk) [inline Week3/src/isInputValid.js]: nice that you added some validation 👍  As you can see this is quite cumbersome to implement. Luckily, there are alternative ways to do this using what is called a schema.  You can red more here http://json-schema.org/learn/getting-started-step-by-step.html

File: Week3/todoAppREADME.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+| path                                | What to delete                                 |
+| ----------------------------------- | ---------------------------------------------- |
+| /lists/_listId_/done                | mark the list as NOT done                      |
+| /users/_userId_                     | delete user                                    |
+| /lists/_listId_                     | delete list                                    |
+| /items/_itemId_                     | delete item(todo)                              |
+| /reminders/_reminderId_             | delete reminder                                |
+| /lists/_listId_/items/_itemId_/done | remove item(todo) in list from completed todos |
+| /lists/_listId_/items/_itemId_      | remove item(todo) from the list                |
```
Comment (gajduk) [inline Week3/todoAppREADME.md]: this is the r rest way to delete associations 👍

**Comment by suh3yb** [comment]
Thank you very much @gajduk for your time and feedback. I will read about project structure and schema, and try to apply them to my future projects.
Best regards

**Pull request closed without merging**